specs/altair/sync-protocol.md

```diff
@@ -101,50 +104,83 @@ def get_subtree_index(generalized_index: GeneralizedIndex) -> uint64:
+    # to accept. If a finalized header is present, it's the finalized header,
+    # otherwise it's the attested header
+    if update.finalized_header != BeaconBlockHeader():
+        return update.finalized_header
+    else:
+        return update.attested_header
+```
+
+### `get_safety_threshold`
+
+```python
+def get_safety_threshold(store: LightClientStore) -> uint64:
+    return max(
+        store.previous_max_active_participants,     
+        store.current_max_active_participants
+    ) // 2
+```
+
 ## Light client state updates
 
-A light client maintains its state in a `store` object of type `LightClientStore` and receives `update` objects of type `LightClientUpdate`. Every `update` triggers `process_light_client_update(store, update, current_slot)` where `current_slot` is the current slot based on some local clock.
+A light client maintains its state in a `store` object of type `LightClientStore` and receives `update` objects of type `LightClientUpdate`. Every `update` triggers `process_light_client_update(store, update, current_slot)` where `current_slot` is the current slot based on some local clock. `process_slot_for_light_client_store` is processed every time the current slot increments.
+
+#### `process_slot_for_light_client_store`
+
+```python
+def process_slot_for_light_client_store(store: LightClientStore, current_slot: Slot) -> None:
+    if current_slot % SAFETY_THRESHOLD_PERIOD == 0:
+        store.previous_max_active_participants = store.current_max_active_participants
+        store.current_max_active_participants = 0
+```
 
 #### `validate_light_client_update`
 
 ```python
-def validate_light_client_update(snapshot: LightClientSnapshot,
+def validate_light_client_update(store: LightClientStore,
                                  update: LightClientUpdate,
+                                 current_slot: Slot,
                                  genesis_validators_root: Root) -> None:
-    # Verify update slot is larger than snapshot slot
-    assert update.header.slot > snapshot.header.slot
+                                 
+    # Verify update slot is larger than slot of current best finalized header
+    active_header = get_active_header(update)
+    assert current_slot >= active_header.slot > store.finalized_header.slot
 
     # Verify update does not skip a sync committee period
-    snapshot_period = compute_epoch_at_slot(snapshot.header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
-    update_period = compute_epoch_at_slot(update.header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
-    assert update_period in (snapshot_period, snapshot_period + 1)
+    finalized_period = compute_epoch_at_slot(store.finalized_header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    update_period = compute_epoch_at_slot(active_header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    assert update_period in (finalized_period, finalized_period + 1)
 
     # Verify update header root is the finalized root of the finality header, if specified
```

Comment:
This is no longer true. It would cause confusion.

Suggested:
# Verify that the `finality_header` is the expansion of `update.attested_header.state_root.finalized_checkpoint.root`